x/rollapp/keeper/invariants.go

```diff
@@ -256,3 +261,48 @@ func RollappFinalizedStateInvariant(k Keeper) sdk.Invariant {
 		), broken
 	}
 }
+
+// LivenessEventInvariant checks for all rollapps that the liveness event height, if any, is accurate,
+// in that there is actually an event stored at that height. Moreover, there should not be any events
+// stored which don't correspond to a liveness event height.
+func LivenessEventInvariant(k Keeper) sdk.Invariant {
+	return func(ctx sdk.Context) (string, bool) {
+		var (
+			broken bool
+			msg    string
+		)
+		rollapps := k.GetAllRollapps(ctx)
+		for _, ra := range rollapps {
+			if !ra.LastStateUpdateHeightIsSet() {
+				continue
+			}
+			events := k.GetLivenessEvents(ctx, &ra.LivenessEventHeight)
+			if !slices.ContainsFunc(events, func(e types.LivenessEvent) bool {
+				return e.RollappId == ra.RollappId
+			}) {
+				broken = true
+				msg += fmt.Sprintf("| rollapp stored event but it was not found in queue: rollapp: %s: event height: %d", ra.RollappId, ra.LivenessEventHeight)
+			}
+		}
+		for _, e := range k.GetLivenessEvents(ctx, nil) {
```

Comment:
after having the check before, i think u can check for rollapp uniquness in the liveness events.
a rollapp should have **single** liveness event

Reply:
done